[MDB IGNORE][IDB IGNORE]Adds back the disposal recycler, and updated it

PsyCommando opened the pull request:
<!-- We recommend to check the contributing page before opening pull requests. -->
<!-- https://github.com/NebulaSS13/Nebula/blob/dev/CONTRIBUTING.md -->

## Description of changes
Adds the disposal crusher back in. Updated the older code to the current codebase. 
And made the machine produce dust piles instead of fully ready to use material sheets. So they have to be processed again in a smelter or something equivalent before being used again.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why and what will this PR improve
Makes processing trash worthwhile and simple. Also handles almost everything that fabs won't eat. Which is a lot of things.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Authorship

<!-- Describe original authors of changes to credit them. -->

## Changelog

:cl:
tweak: Added the disposal recycker machine. 
/:cl:

<!-- Replace `prefix` with one of tags below. You can add more tags for multiple changelog entries.

- bugfix
- balance
- tweak
- soundadd
- sounddel
- rscadd
- rscdel
- imageadd
- imagedel
- maptweak
- spellcheck
- experiment
- admin

-->

quardbreak committed:
sanitizing: ports sanitize updates from Tau Ceti Station

PsyCommando committed:
Adds base recycler code and dust

PsyCommando committed:
Bunch of fixes

* Made thrown items processed by the recycler.
* Cleaned up things a bunch and used a timer for handling safe mode.
* Added saved vars.
* Fixed dust pile icon.
* Made recycler circuit a design in the circuit imprinter.
* Fixed machine destruction not releasing content.

PsyCommando committed:
Fixed missing type

* That actually came from a fix for the badly typed graphite stacks defined in persistence that I forgot to split off. But anyways.

NataKilar commented on mods/persistence/modules/materials/material_sheets.dm:
```diff
@@ -1,9 +1,39 @@
+	max_icon_state = "dust-max"
+	stack_merge_type = /obj/item/stack/material/dust
+	max_amount = 200
+/obj/item/stack/material/dust/list_recipes(mob/user, recipes_sublist)
```
Nit: Spacing between the lines of the proc definitions would be appreciated.

PsyCommando replied:
Will do!

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+///////////////////////
+// Blacklist
+///////////////////////
+var/global/list/recycler_item_blacklist = list(
```
This could be a static list on the recycler itself.

PsyCommando replied:
That's fair.

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+	)
+
+	var/bloodied 				= FALSE		//Whether we show the sprite with blood or not
+	var/tmp/grinding			= FALSE 	//Whether we're currently busy grinding something down
```
Why the tmp variables? While I think there might be some very old usages of issaved in our saving system, since we're working off a whitelist these really shouldn't be necessary, unless there's something I'm missing.

PsyCommando replied:
Its good practice to mark runtime vars tmp, and constants const. Also the code still checks issaved, on vars even in the white list. Plus it helps prevents people from adding it to the saved var list when they really don't need to.

NataKilar replied:
Because /tmp/ isn't used elsewhere, I disagree that this is good practice.  It only confuses things and will make saving the var fail silently if someone does add it down the line (even erroneously) . I think really that issaved() check should be removed.

I'd be more willing to formalize use of /tmp/ if we could modify upstream variables to use it. Otherwise it's just a confusing standard that could only ever be applied in the persistence modpack.

PsyCommando replied:
Just look at the nebula code. A lot of new code they made respects proper /tmp use. 

And I'm gonna be blunt here, I'm not sure you really know what you're talking about here? We don't wanna remove is_saved. A lot of variables in base byond code aren't actual variables, and if some clown decides to add them to the saved list without that check we could easily break things and spend hours trying to figure out why... This is a particularity of the language we're using, we can't just ignore it and pretend everything is fine.

The solution to fix silent failing is to actually make checks return errors, which most of the checks in the current saving code never do right now for some reasons. To remove the actual checks instead is nonsensical.

NataKilar replied:
Using /tmp/ for runtime vars is fine. I wasn't aware of this use upstream. To be honest, I'm still a little tepid on using the built in issaved() proc but it's probably not bad practice. The silent erroring should be fixed at some point, but outside of this PR.

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+
+	var/bloodied 				= FALSE		//Whether we show the sprite with blood or not
+	var/tmp/grinding			= FALSE 	//Whether we're currently busy grinding something down
+	var/tmp/safety_mode 		= 0 		// Temporality stops the machine if it detects a mob
```
Temporarily

PsyCommando replied:
That's a super old typo lol. That's literally the ported comments from the original code XD

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+	var/const/sound_reject 		= 'sound/machines/buzz-sigh.ogg'
+	var/tmp/soundid 			= "Recyclers"
+	var/tmp/datum/sound_token/sound_looping
+	var/eat_dir = WEST						//Fixed directio from where the machine will accept trash
```
Direction

PsyCommando replied:
Direction what?

NataKilar replied:
"Fixed directio" comment, that is. Should have been more clear, sorry.

PsyCommando replied:
Yeah, that's because the sprite doesn't rotate, and is always facing the same way. The spiny bits will always face the same direction, which is gonna confuse people.

NataKilar replied:
Only talking about the missing "n" in direction

PsyCommando replied:
Oh! Gotcha!

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+		return ..()
+	process_input(AM)
+
+/obj/machinery/recycler/Bumped(atom/movable/AM)
```
Maybe add identical behavior for hitby(), so that things can be recycled by throwing without a full conveyor setup.

PsyCommando replied:
They both call process_input?

NataKilar replied:
Not sure how I missed that, my mistake.

NataKilar commented on mods/persistence/modules/recycling/recycler.dm:
```diff
@@ -0,0 +1,383 @@
+
+/obj/machinery/recycler/proc/reject_item(var/atom/movable/AM, var/activate_safety = FALSE)
+	playsound(src.loc, 'sound/machines/buzz-sigh.ogg', RECYCLER_FEEDBACK_SOUND_VOLUME, vary=FALSE, extrarange=8, falloff=3)
+	AM.forceMove(src.loc)
```
Maybe move it to the output side instead?

PsyCommando replied:
That's the ported implementation. If I change that, I'll need to check if it breaks anything.

Andrew-Fall committed:
Update material_sheets.dm

MarinaGryphon committed:
Use SSoverlays for beds/chairs/stools

MarinaGryphon committed:
Fix runtime in 01_species.dm:29

MarinaGryphon committed:
Fix runtime in ghost.dm:63

MarinaGryphon committed:
Fix runtime in ignore.dm:40

MarinaGryphon committed:
Fix runtime in rig.dm:779

MarinaGryphon committed:
Fix runtime in attack_logs.dm:37

Commit:
Standardizing structure icon updates to use SSoverlays.

Commit:
boiling temp depends on pressure formula

Commit:
STP to codex

MistakeNot4892 committed:
Blood types are now handled as decls.

MistakeNot4892 committed:
Species now have distinct blood types.

MistakeNot4892 committed:
Standardized spelling of color for blood.

MistakeNot4892 committed:
Adding transfusion vars to blood types.

MistakeNot4892 committed:
Fixes minor path issue with abstract objects.

MistakeNot4892 committed:
Landmarks are now abstract objects.

MistakeNot4892 committed:
Repaths map_data to abstract instead of abstract/landmark.

MistakeNot4892 committed:
Merge pull request #2238 from PsyCommando/fixing-inventory-hiding

Fixing inventory hiding redux

NebulaSS13Bot committed:
Automatic changelog generation for PR #2240 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2238 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

SolatK committed:
fix possible division by zero

NebulaSS13Bot committed:
Automatic changelog generation for PR #2239 [ci skip]

MarinaGryphon committed:
Update in world count

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

MarinaGryphon committed:
Fix runtime in assembly_computer.dm,29
Runtime in assembly_computer.dm,29: undefined variable
/obj/item/stock_parts/smes_coil/super_io/var/usage_flags

MarinaGryphon committed:
Fix runtime in ftl_jump.dm:122
Ported from Bay.

MarinaGryphon committed:
Fix pipe wrench not having wrench archetype

MarinaGryphon committed:
Fix improper color on plastic curtains

MarinaGryphon committed:
Fix paint sprayer runtimes; improve code

MarinaGryphon committed:
Add missing string punctuation, first pass

MarinaGryphon committed:
Add missing string punctuation, second pass

MistakeNot4892 committed:
Merge pull request #2233 from MarinaGryphon/feature/reboot-hook

Add reboot hook to world/Reboot

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

MistakeNot4892 committed:
Removing SHOULD_CALL_PARENT from structure on_update_icon() procs.

MistakeNot4892 committed:
Allows cosmosharks to update icon properly.

MistakeNot4892 committed:
Expanded text entry field for mind read, added sanitization.

MistakeNot4892 committed:
Cans and bottles now opt out of desc/name customization based on contents.

MistakeNot4892 committed:
Brain damage text is no longer twice the maximum size.

MistakeNot4892 committed:
Wind effects are no longer applied to turfs with no atmosphere.

MistakeNot4892 committed:
Cleans up some uses of the wrong tape type, clarifies taperoll vs tape_roll.

MistakeNot4892 committed:
APCs should play their failure sound only on first failure.

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

MistakeNot4892 committed:
Merge pull request #2247 from MarinaGryphon/nebfixes/12-04-2021

December 4, 2021 Bugfixes

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

MarinaGryphon committed:
Remove continue-to-label in ailment handling

Lohikar committed:
lighting: goto -> do-while

MistakeNot4892 committed:
Merge pull request #2249 from Lohikar/goto-hell

Lighting: Remove goto

quardbreak committed:
keybind: fixes move-upwards

fixes: https://github.com/NebulaSS13/Nebula/issues/2138

quardbreak committed:
doors: fixes empty icon_state on spawn

fixes: https://github.com/NebulaSS13/Nebula/issues/2250

MistakeNot4892 committed:
Merge pull request #2228 from MarinaGryphon/feature/ailment-expansion

Expand and refactor ailment treatment

MistakeNot4892 committed:
Merge pull request #2251 from quardbreak/key-door

keybind and door fixes

quardbreak committed:
rack: fixes unanchored state

quardbreak committed:
sstimer: fix potential bucket corruption in timer reset_buckets

quardbreak committed:
paper: replaces find and copy text procs with char variant

MistakeNot4892 committed:
Merge pull request #2256 from quardbreak/rack-anchored

rack: fixes unanchored state

quardbreak committed:
test.yml: SpacemanDMM suite 1.7.1 and BYOND 514.1572

quardbreak committed:
pr-template: cut empty lines, author name override comment

quardbreak committed:
screen-movable: fixes loc offset on assign

afterthought2 committed:
Merge pull request #2258 from quardbreak/ppcptxt

paper: replaces find and copy text procs with char variant

afterthought2 committed:
Merge pull request #2259 from quardbreak/sdmm-171

test.yml: SpacemanDMM suite 1.7.1 and BYOND 514.1572

afterthought2 committed:
Merge pull request #2260 from quardbreak/pr-template-less

pr-template: cut empty lines, author name override comment

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

quardbreak committed:
sstimer: adds as anything for loop

Vallat committed:
working wide-screen parallax

quardbreak committed:
dbban: adds database based ban panel to verbs list

Vallat committed:
Update code/_onclick/hud/skybox.dm

Vallat committed:
weird runtime error fix

quardbreak committed:
ventcrawl: diagonal direction no longer breaks movement

afterthought2 committed:
Merge pull request #2263 from Vallat/parallax

Reworking skybox scaling to take client scale into account.

afterthought2 committed:
Merge pull request #2257 from quardbreak/sstimer-bucket-corruption

sstimer: fix potential bucket corruption in timer reset_buckets

afterthought2 committed:
Merge pull request #2248 from MistakeNot4892/fixes

Various fixes.

Gaxeer committed:
fix: fix looping APC sound

afterthought2 committed:
Merge pull request #2265 from quardbreak/dbban-panel

dbban: adds database based ban panel to verbs list

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2272 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

Gaxeer committed:
fix: grabbed dead mobs now able to move after the grabber

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

afterthought2 committed:
Merge pull request #2273 from Gaxeer/deadmobs

fix: grabbed dead mobs now able to move after the grabber

NebulaSS13Bot committed:
Automatic changelog generation for PR #2273 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

afterthought2 committed:
Merge pull request #2268 from quardbreak/ventcrawl-diagonal

ventcrawl: diagonal direction no longer breaks movement

afterthought2 committed:
Merge pull request #1830 from quardbreak/sanitize-new

Revisits sanitize, ports sanitize from TauCeti

PsyCommando committed:
Made most botany things save properly (#262)

* Made most botany things save properly

* Added reagents to saved vars because it apparently wasn't there for some reasons and nothing saved their reagents contents.
* Made smartfridges save their contents
* Made seed storage save their contents
* Added a lot of saved vars for hydro trays since they weren't saving anything relevant besides static lists for some reasons.
* Produces now don't disappear on load, and have their seed properly loaded.
* Seeds are now only saved if they're not one of the ones defined at roundstart.
* Genes on botany disks are now saved and seem to match the proper gene after load.
* Vines should now save properly. (Haven't tested those much)
* Prevented bottles from respawning their contents on load.
* Overridden the create_reagents proc to skip over spamming that the object already has reagents, since its almost always the case now. The message is more useful upstream anyways.

* Applied fixes

* Fix some time values not being restored properly on load

NataKilar committed:
Fixes migration schema (#261)

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

Commit:
Fix air alarm sensor settings

Commit:
Add splash_turf/splash_area helpers

Commit:
Fix and refactor seed splattering
Fixes a bug with incorrect vars being passed to
`/obj/effect/decal/cleanable/fruit_smudge/Initialize()`.

Commit:
Fix mushroom vines/spores on throw_impact

noelle-lavenza committed:
Fix runtime in humans.dm:112

noelle-lavenza committed:
Fix antagonist rolling

PsyCommando committed:
Cut up and reorganized the huma init code a bit (#2184)

* Cut up and reorganized the huma init code

My first idea didn't turn out as planned, so I've toned down the scope of the changes I wanted to make to something more manageable. Factories got messy and turned into a pain in the ass to use and a ton of barely useful subtypes.

* Moved code outside of Initialize into 2 procs, so they can be overriden or completely replaced by subclasses.
* Changed the set_species proc so it doesn't handle most of initializing the mob anymore and can be expected to do the bare minimum it should do.
* Moved the extra processing the set_species proc did to other procs so they can be overriden or skipped alltogether if neccessary.
* Moved verbs human mob verbs to human_verb.dm since the human.dm file is a bit big already.
* Moved the define for /mob/living/empty_stomach() to the living.dm file since it was randomly hiding in the human code and kinda bugged me.

-> Changed some of the code for changelings and the change_species proc with some untested code for now. Will need more work most likely.

* Moved human and living procs to the proper files

So hopefully anyone looking for stuff in the future isn't gonna have to search for thing being under the wrong file...

* Renamed dna parameter in several human proc + added comments

* Added check to avoid replacing organs if we already have organs when running human init

* Got rid of the Login() call that was in the human init code and replaced it with a new proc hud_reset()

* Added a sanity check to IsSynthetic for human mobs so when there's no organs yet it doesn't break

If it was called early it would consider the mob with no organs yet to be a cyborg, and cause all sort of weird issues. Could definitely use an improvement..

* Modified organ init to be more compatible with save loading + generalized icon handling a bit

* Fix runtimes with adherents + fix robotize trying to create a cell before the chest actually existed

I think the code in standard.dm I removed is old code for handling FBP specifically. I removed it for now considering it caused a lot of problems and I can't find a good way to have it do its thing properly without adding more order depended stuff to human init.

* merge

* WOW CI not on 514 yet, what a shame!

* holder type fix

* shutup ci its not a mob typepath

* Fix blanks runtiming

* Missing material_key parameter added to organ Initialize procs

* The second parameter passed to items is a material_key, and organs didn't have it listed in their parameters, so I added it to prevent future confusion.
* Fixes issues with unit tests and etc initializing the class with a material key causing runtimes.

* fix runtime on implanting a mob with no mind

* Rework of the organ init and installation/uninstallation code

* Added a more flexible way of telling if organs are vital or not.
* Renamed the remove/replaced procs on organs and split them to install/uninstall and on_removal/on_replacement. install/uninstall handles properly registering the organ and etc, while on_removal/replacement handles effects that would happen on those cases. EX: on_removal for sabotaged prothetics has the check for sabotaged organs exploding on removal.
* Added a proc to encapsulate changing the organ state to ORGAN_CUT_AWAY since external prothetics should not have the flag turned on.
* Modified slightly the intended role of the /decl/species/proc/create_organs proc. So that it doesn't fully replace organs by default and only remove those that are incompatible. The old functionality is still there and toggled via argument.
* Changed the order in which organ addition/removal is handled. Now the owner is at the top of the chain so overrides and handling propeerly vital organs can be achieved. And it also makes it way easier to validate icon updates and etc without risking organs calling updates at the wrong time.
* WIP Since all internal organs only ever have 4 icon states across all of them, moved them to a variable for each in the base class so there's no need to redefine update_icon for all organs with more than one state.
* Modified the logic for ghostizing mobs on brain install if they have a client. So that it only happens when the brain organ has an inhabited brainmob to avoid clients getting kicked out to a ghost when doing a mob transformation.
* Added a spawn for handling the interactive part of installing a vox stack organ on someone, so that doesn't cause issues during organ init.

* woops

* Several fixes

* Fixed issue with create_missing_organs where existing organs would be kept even when fully replacing was true.
* Properly remove references to organs on destruction and when using create_missing_organs.
* Skip a lot of unnecessary icon updates now when destroying organs and mobs.
* Moved organ installation on initialize to an overridable proc since internal_organs have a different way of installing themselves.
* Fixed transforming to a species with a posibrain disconnecting the client because of the weird brainmob logic for posibrains.
* Added vital organ lists for all species and added vital limbs to the list.
* Changed some of the checks for vital organs to use the species's proc when applicable. Some cases weren't really applicable though, since the species' proc is circumstantial.

* typo fixes in comments + removed a testing() I forgot to remove

* Applied some of the requested fixes

* Applied a few fixes

* Made the vox stack replacement prompt waitfor = false, instead of running a timer.
* Removed the code that would automatically install an organ if it was inside a carbon mob on creation, since its not been in use for ages, and causes problems with save/load downstream anyways when dealing with "detached" organs and etc..
* Removed a useless call to uninstall() on appendices.
* Some calls to the DNA constructor had a null as parameter, so removed that.
* Changed one legacy use of set_species in admin tools to use change_species instead.

* Ported organ access and vars change from loaf's PR, and made some changes

* Both in internal and external organs are accessed through "get_organ()". To get the whole lists you can now use "get_internal_organs()", "get_external_organs()" or "get_organs()".
* Had to change the way loaf wrote get_organ(), because he would call check_zones on all organs passed, which would return the incorrect organs and cause a lot of issues. Left using check_zone() to the discretion of the calling proc.
* Changed his code so installation goes through the parent mob, and install/uninstall effects are once more separate.
* Put the base implementation of organ procs in mob/living instead of mob like in loaf's code since there's not really much point to get organs on a non-living mob. (Left the calls needed by grabbing procs in there)
* Renamed organ procs "install" and "uninstall" to "do_install" and "do_uninstall", to try to avoid suggesting those handle the entire installation process.
* Renamed "on_removal" and "on_replacement" to "on_remove_effects" and "on_add_effects" to better reflect what the procs are intended to do.
* Made a proc to handle organ deletion for all subclasses of mob/living/carbon. This is because, humans add lists keeping organ refs that really have to be cleared at the time we delete the organs.
* Added a "is_internal" check to organs so a few type checks can be avoided down the line.
* Removed a typecheck in /obj/item/external/die() where it would check if something was a groin or chest type to override die() behavior. Overriden the proc for both subtypes instead, so less chances for future errors.
* Remove prosthetics hack that would change the prosthetic's species to the owner's species each times it was installed, since its essentially a bug, and prosthetics have already a species they're created with that shouldn't be changed randomly.

* Removed unused DNA procs without ROCKDTBEN's permission

* Fixed bug with gibbing mobs

* Intermittently, the mob would be deleted before the timer would run, and cause a runtime. Used QDEL_IN instead on both the animation and mob separately so there's no dependency on the mob still existing.

* Fix insectoids not having a parent organ for their egg_sacs..

* Removed reduncdant holder_type = null

* commented out debug output for now

* Removed more debug output..

* Removed bad leftover duplicate of /mob/living/carbon/get_organ(var/zone)

* I hate bugs

* Fix gynes adding internal egg sacs as an external organ

* Small tweaks

* Added a check to only refresh modular limb verbs when we're installing/removing an external limb.
* Added missing parameter to remove_organ declaration
* Re-ordered PRINT_STACK_TRACE since it was making some code unreachable
* Removed leftover commented debug traces.

NebulaSS13Bot committed:
Automatic changelog generation for PR #2184 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

noelle-lavenza committed:
Fix runtime in antagonist selection

noelle-lavenza committed:
Fix brute healing medications

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2281 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

noelle-lavenza committed:
Fix exoplanets never spawning with gas

noelle-lavenza committed:
Prevent internals forming a vacuum under pressure

afterthought2 committed:
Merge pull request #2283 from MarinaGryphon/tweak/autoasphyxiation

Prevent internals forming a vacuum under pressure

afterthought2 committed:
Merge pull request #2282 from MarinaGryphon/fix/exoplanet-atmos

Fix exoplanets never spawning with gas

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

noelle-lavenza committed:
Implement changes from code review

NebulaSS13Bot committed:
Automatic changelog generation for PR #2278 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

noelle-lavenza committed:
Fix character setup + change_species

kelgaras committed:
say no to window murder

say no to window murder

comma committed:
Fixes #2216

Special icon was only drawn if there's buckled mob, but shuttle chair had one for both states.
Only other chair to use special icons (cap chair) has one that works for both cases too, so just axed buckled check. Code in the block already handles layering in case of buckled/not anyway.

comma committed:
Fixes #2211

Updates bordercheck flag when window direction is set

kelgaras committed:
You can force move mob on catwalk in passive grab

You can force move mob on catwalk in passive grab

comma committed:
Fixes #2004

It seems to be mostly caused by relays never reconnecting and therefore entire decks losing network connectivity forever.
Now if network is shutting down, relays will start trying to reconnect (every 30 seconds), so if router goes up, relays will follow soon.

PsyCommando committed:
Refactored fabricators to be more flexible and made their UI prettier

* Split across several procs the passing of UI data the nano template, so they can be overridden and disabled as needed in sub classes.
* Added support for shared templates to the fabricator, so shared template functions and styles can be shared across files easily.
* Added support for creating sub class specific nano templates for fabricators.
* Split off the network header to its own shared template for future use.
* Remade the layout and appearance of the fabricator UI to be a lot more readable and compact.

PsyCommando committed:
Fixed nano bugs

* Fixed nano creating a miilion poppup to tell you a template failed compiling.
* Fixed the link to the nanoui doc in the TemplateGuide.txt file.

noelle-lavenza committed:
Fix new linter switch warnings

afterthought2 committed:
Merge pull request #2285 from keIgaras/glass

say no to window murder

NebulaSS13Bot committed:
Automatic changelog generation for PR #2285 [ci skip]

afterthought2 committed:
Merge pull request #2288 from comma/routing

Fixes #2004

NebulaSS13Bot committed:
Automatic changelog generation for PR #2288 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

kelgaras committed:
Open door sprite now renders correctly

Open door sprite now renders correctly

NebulaSS13Bot committed:
Automatic changelog generation for PR #2286 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2292 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

kelgaras committed:
Change the location of the ui_storage

Change the location of the ui_storage

noelle-lavenza committed:
Fix logic error in is_default_limb

noelle-lavenza committed:
Remove bad update_appearence_descriptors call

noelle-lavenza committed:
Fix specie/appearence typos

noelle-lavenza committed:
Refactor syndiebeacon code slightly

afterthought2 committed:
Merge pull request #2287 from comma/chairfix

Couple small fixes

NebulaSS13Bot committed:
Automatic changelog generation for PR #2287 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

SierraKomodo committed:
Allow AltClick to remove ID from wallets

noelle-lavenza committed:
Refactor more syndiebeacon code

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into fabricator-ui-refactor

afterthought2 committed:
Merge pull request #2284 from MarinaGryphon/hotfix/charsetup

Fix character setup + change_species

NebulaSS13Bot committed:
Automatic changelog generation for PR #2293 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NataKilar committed:
Fixes wall property transfers

NebulaSS13Bot committed:
Automatic changelog generation for PR #2296 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

kelgaras committed:
sofa adding

SierraKomodo committed:
Don't display `hear_sleep` to deaf people

NebulaSS13Bot committed:
Automatic changelog generation for PR #2297 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2298 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

hermaplusplus committed:
Lazyaccesses reagent_amounts when filling grown foods

PsyCommando committed:
Improvements and clean up

* Decided to keep fabricators as close to their original functionalities as possible to reduce chances for bugs and etc..
* Bioprinters and Robotics printers have only a single config entry to be set. Species for robotics, and dna for bioprinter.
* Category filter is now only displayed when there's more than a single category of designs available.
* Added filled preset for bioprinter.
* Added overrides for subclass of fabricators to grey out printing items based on extra checks, and straight up not queue things.

PsyCommando committed:
Bioprinter desc fix

PsyCommando committed:
Fixed passing a manufacturer path to robotize runtiming

PsyCommando committed:
Fully Si Compliant now!

afterthought2 committed:
Merge pull request #2299 from hermaplusplus/food-chemfill-fix

Lazyaccesses reagent_amounts when filling grown foods

NebulaSS13Bot committed:
Automatic changelog generation for PR #2299 [ci skip]

PsyCommando committed:
Removed extra is_functioning checks

PsyCommando committed:
re-ordered fabs OnTopics so any returned TOPIC_REFRESH is no longer completely ignored

PsyCommando committed:
Removed extra dna nullcheck

PsyCommando committed:
Modified fab build order

* Added procs to access custom data
* Made calls to recipe.build take a build order as second param.
* Added handy constructor for build orders

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

PsyCommando committed:
Organs now delete themselves when they should be! Amazing

PsyCommando committed:
Renaming the brain during removal from mob now happens before its removed, so it actually happens

afterthought2 committed:
Merge pull request #2300 from PsyCommando/fix-delete_organs-not-deleting-organs

Fix delete_organs() not actually deleting organs

afterthought2 committed:
Merge pull request #2301 from PsyCommando/Fix-removed-brains-name

Fix de-braining not renaming the brain

NebulaSS13Bot committed:
Automatic changelog generation for PR #2300 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation for PR #2301 [ci skip]

PsyCommando committed:
Fixed species name not updating on the robotic fab UI

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

afterthought2 committed:
Merge pull request #2289 from PsyCommando/fabricator-ui-refactor

Fabricator UI refactor

NebulaSS13Bot committed:
Automatic changelog generation for PR #2289 [ci skip]

PsyCommando committed:
Fix re-installing an organ during surgery on the same mob

The organ wouldn't get cleared from the mob's organ lists properly, so fixed that.
* Instead of slapping in some hardcoded snowflake code for the mmi_holder I've made some procs to override the behavior of organ detach during surgery.
* Removed the contents of the vital_organs list in the species. Since removing some organs would cause unintended death and brain damage.

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

PsyCommando committed:
Fixed dismembering

Dismembering would not transfer all internal organs and child limbs to the severed parts if there were several layers of children.

PsyCommando committed:
Fixed stumps getting dropped on physically_destroyed

PsyCommando committed:
Added a log message when installing an organ with ORGAN_CUT_AWAY

It would silently skip some processing before. Added it for future error tracing.

PsyCommando committed:
Removed unnecessary LAZYREMOVE

PsyCommando committed:
Removed redundant remove_organ

PsyCommando committed:
Quick fix for godmode not preventing explosion damage

PsyCommando committed:
Use proper maxhealth constant on updatehealth instead of 100

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

PsyCommando committed:
Properly add stumps to their slot on dismember

Stumps were assuming they'd be installed during constrution, which isn't the case anymore, since installing organs at init caused a lot of ordering issues before the refactor. So installed it post-construction, like everything else is.

PsyCommando committed:
Fix stumps coming out again

Turns out, the way organ lists get cleared on destroy means that its really unreliable to rely on the stumps being in the organ lists to not get dropped on gib. Since if someone reorders things a bit, the stumps will get dumped again and nobody will know why.
Soo, I went back with my previous solution of excluding stumps from the contained organs that get dropped in dump_contents. Since its way more reliable.

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

PsyCommando committed:
Make sure gib() doesn't throw around stumps by default

PsyCommando committed:
Several fixes to external organ handling, and surgeries

* Realized external organs never ended up in the implant table in the old code, and it broke things, soo made sure they never end up in there anymore. Its annoying because I can't treat both organ types the same way however. But human icons and surgery are made that way.
* Used add_organ and remove_organ instead of accessing mob's internals during surgery. Attaching and detaching are fully supported.
* Made sure stumps and other organs we don't want to exist outside the mobs get deleted on drop.
* Added is_droppable() proc to tell what organ cannot be dropped out of the mob since the organ amputation flag is not a good indicator.(stumps need amputation flag on, but not the root limb. And both cannot be dropped)
* Made the mmi_holder turn into an mmi when dropInto() is called. Since its mega hacky, and I wasn't gonna change everything for a single weirdly implemented proxy organ.
* Added to amputation wounds the actual limb that got amputated, so clearing wounds during the surgery is working, instead of randomly deleting the last wound added.
* Reordered the external organ destruction, so that refs are properly removed, and things deleted.
* Added a lot of checks to make sure we handle the new use case of adding detached organs via add_organ and do_install.
* Gibbing now makes sure its not going to throw around non-droppable organs.
* Removed surgery specific mob procs.

PsyCommando committed:
Added unit test making sure nothing drops stumps

* Since its been a reoccurring problem, might as well add a test for it to prevent people re-introducing it again.

PsyCommando committed:
Updated "in world" use counts in code quality checks

NebulaSS13Bot committed:
Automatic changelog generation for PR #2305 [ci skip]

NebulaSS13Bot committed:
Automatic changelog generation [ci skip]

PsyCommando committed:
Merge branch 'dev' of https://github.com/NebulaSS13/Nebula into upstream_merge

PsyCommando committed:
Make things actually compile

PsyCommando committed:
Fixed starter map being weird

Updated starter map's map_data type path and ladmark type paths to the new paths.

PsyCommando committed:
Merge branch 'organ-removal-surgery-fix' of https://github.com/PsyCommando/Nebula-1 into upstream_merge

PsyCommando committed:
Fix x flag on check_path.sh

PsyCommando committed:
perm fixed pls?

PsyCommando committed:
trying to remove random leftover changelog crap

PsyCommando committed:
removed extra goto use in tests

NataKilar committed:
Merge pull request #271 from PsyCommando/upstream_merge

Upstream merge

PsyCommando committed:
Update code/modules/organs/external/_external.dm

Co-authored-by: Jade Lavenza <101909198+jade-lavenza@users.noreply.github.com>

PsyCommando committed:
Fix is_droppable typo

PsyCommando committed:
Update code/modules/species/species.dm

Co-authored-by: Jade Lavenza <101909198+jade-lavenza@users.noreply.github.com>

PsyCommando committed:
Apply suggestions for stumps unit test

Co-authored-by: Jade Lavenza <101909198+jade-lavenza@users.noreply.github.com>

PsyCommando committed:
Update code/unit_tests/organ_tests.dm

Co-authored-by: Jade Lavenza <101909198+jade-lavenza@users.noreply.github.com>

PsyCommando committed:
Fix the suggested fix

PsyCommando committed:
Merge branch 'organ-removal-surgery-fix' of https://github.com/PsyCommando/Nebula-1 into upstream_merge

PsyCommando committed:
Add extra checks to carbon mob init

Since we're loading those things, the base code should make sure they're null before creating new reagents.

PsyCommando committed:
Remove check to prevent setting a species if the same one is already set

We have to allow setting the same species again, so that the proper init code can be run on save load.

PsyCommando committed:
Create blood only if its null on human mobs

Since we're loading from save, "vessel" may be non null when loading, and its content shouldn't be reset.

PsyCommando committed:
Fix carbon saving/loading

* Made sure organs are installed in place on save load, so nothing breaks.
* Changed saved vars so they match the new way organs work.

PsyCommando committed:
Organs now save species and bodytype

PsyCommando committed:
Make humans load from save properly with the organ/init changes

PsyCommando committed:
Add missing saved  var for mob/living

PsyCommando committed:
Made stacks properly update languages and verbs on load

* Changed stack init code to be less redundant.
* Moved verb and language setup to the do_install/do_uninstall procs, since language lists tend to be short-lived.
* Removed human hack to install cortical language if a human has a stack organ.
* Added a macro to clear a custom saved var, for coherence's sake.
* Renamed CLEAR_SV to CLEAR_ALL_SV

PsyCommando committed:
Added missing language to human cultures

Seems like nobody noticed all human culture didn't have a language since it manually set in like 3 different spots..

PsyCommando committed:
Added additional runtime throws when organs are missing a bodytype or species and trying to do gfx updates

PsyCommando committed:
Added proper saved vars on blood decals

Blood decals are needed for forensics, so made them save properly. Dried out puddles don't save, since they become useless once dry.

PsyCommando committed:
Made the default var warning clearer

Things that get saved and don't have defined saved print this debug warning. Just clarified that it did in the message.

PsyCommando committed:
Save a few human appearance vars separately from dna

Since those values can be changed independently from dna in theory saved them separately, and loaded them after dna is set. Also, prevent dna code weirdness from causing weirdness. (AKA it turns values that are 0 into values that are 1...)

PsyCommando committed:
Removed the useless age field from the chargen computer

PsyCommando committed:
Fix missing args in recursive calls

PsyCommando committed:
Fixed wounds and blood not reloading correctly

* Rewrote handheld item loading from save, so it works.
* Removed redundant proc calls.
* Removed blood overlay from saved vars, since its cached, and its pointless to save it.
* Added coating to saved vars for items instead of blood overlay
* Made blood overlay recreated on load from save.
* Properly update wounds from save after load.

PsyCommando committed:
Fix held items not displaying on hud after load from save

PsyCommando committed:
Made hairstyle and facial hair style save separately from DNA

Those are oftend changed to something else than the value in the DNA too.

NataKilar committed:
Merge pull request #274 from PsyCommando/Remove-age-from-chargen

Removed the useless age field from the chargen computer

PsyCommando committed:
Removed unnecessary check when updating limbs icon

PsyCommando committed:
Revert "Remove check to prevent setting a species if the same one is already set"

This reverts commit d3975141fd44f38908bb78a618a8bdc5441c0ac5.

PsyCommando committed:
Reworked human mob species init so we don't need to change set_species()

PsyCommando committed:
Moved species and bodytype saved vars under carbon where they are defined

PsyCommando committed:
Didn't mean to push that line..

NataKilar committed:
Merge pull request #273 from PsyCommando/human-saving-rework

Human saving rework

PsyCommando committed:
Merge pull request #272 from PsyCommando/upstream_merge

Extra fixes for #271

PsyCommando commented:
Converting to draft for now, since its gonna need some fixing up to bring it back up to date.

PsyCommando committed:
Revert "Fixed missing type"

This reverts commit afabe8a9ee1660d0a581b6e747734535ba814077.

PsyCommando committed:
Merge branch 'add-recycler' of https://github.com/PsyCommando/Nebula into add-recycler

AffectedArc07 commented:
Please put `[MDB IGNORE]` and `[IDB IGNORE]` into your PR title.

You crashed the service for almost 10 minutes.
![image](https://user-images.githubusercontent.com/25063394/166159294-d372eeee-1f80-44f2-ba82-17f7e19b9200.png)

PsyCommando commented:
> Please put `[MDB IGNORE]` and `[IDB IGNORE]` into your PR title.
> 
> You crashed the service for almost 10 minutes. ![image](https://user-images.githubusercontent.com/25063394/166159294-d372eeee-1f80-44f2-ba82-17f7e19b9200.png)

Sorry, been trying to salvage this very old pr, aaaand I think its probably not gonna work out...
